src/commands/ReminderCommand.ts

```diff
@@ -61,32 +72,58 @@ class ReminderCommand extends Command {
 
 		switch (subcommand) {
 			case "set": {
-				//default 10 minutes
-				let minutes = interaction.options.getInteger("minutes") || 0;
+				const minutes = interaction.options.getInteger("minutes") || 0;
 				const hours = interaction.options.getInteger("hours") || 0;
 				const days = interaction.options.getInteger("days") || 0;
+				const months = interaction.options.getInteger("months") || 0;
+
+				const dateIso = Date.parse(interaction.options.getString("date-iso") || "error");
+				const dateUnix = new Date(
+					parseInt(interaction.options.getString("date-unix") || "error") * 1000
+				).getTime();
+
 				const message = interaction.options.getString("message") || "";
-				const callAll = interaction.options.getBoolean("callall") || false;
-				let milliseconds = (60 * minutes + 60 * 60 * hours + 60 * 60 * 24 * days) * 1000;
-				if (milliseconds == 0) {
-					// defaults to 10 minutes
-					milliseconds = 10 * 60 * 1000;
-					minutes = 10;
+				const additionalPing = interaction.options.getMentionable("additional-ping") as
+					| GuildMember
+					| Role
+					| null;
+
+				const milliseconds = (minutes + 60 * hours + 60 * 24 * days + 60 * 24 * 30 * months) * 1000 * 60;
+				let timestamp: number;
+				if (dateIso) {
+					timestamp = dateIso;
+				} else if (dateUnix) {
+					timestamp = dateUnix;
```

Comment:
often unix timestamps are in seconds meanwhile in javascript they are milliseconds, could probs check the magnitude of the value and multiply accordingly

Reply:
oops this was done above